### PR TITLE
minor api changes for ferminet layers

### PR DESCRIPTION
**Pull request by shaipranesh2**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/tests/test_layers.py [hunk 1]
```diff
@@ -1283,24 +1283,25 @@ def test_FerminetElectronFeature():
     "Test for FerminetElectronFeature layer."
     electron_layer = dc.models.torch_models.layers.FerminetElectronFeature(
         [32, 32, 32], [16, 16, 16], 4, 8, 10, [5, 5])
-    one_electron_test = torch.randn(8, 10, 4 * 4)
-    two_electron_test = torch.randn(8, 10, 10, 4)
+    one_electron_test = torch.randn(8, 10, 4 * 4).double()
```
Comment (rbharath) [inline deepchem/models/tests/test_layers.py:1286]: Why are we doubling these? Can you add comments that explain the purpose?

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -4942,26 +4942,34 @@ def __init__(self, n_one: List[int], n_two: List[int], no_of_atoms: int,
         # Initializing the first layer (first layer has different dims than others)
         self.v.append(
             nn.Linear(8 + 3 * 4 * self.no_of_atoms, self.n_one[0], bias=True))
-        #filling the weights with 1e-9 for faster convergence
-        self.v[0].weight.data.fill_(1e-9)
-        self.v[0].bias.data.fill_(1e-9)
+        #filling the weights with 2.5e-7 for faster convergence
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4945]: Why are we making this change? What is the purpose of the doubling?

File: deepchem/models/torch_models/layers.py [hunk 3]
```diff
@@ -5000,21 +5008,17 @@ def forward(self, one_electron: torch.Tensor, two_electron: torch.Tensor):
                     two_electron[:, i, self.spin[0]:, :], dim=1)
                 f: torch.Tensor = torch.cat((one_electron[:, i, :], g_one_up,
                                              g_one_down, g_two_up, g_two_down),
-                                            dim=1)
+                                            dim=1).double()
                 if l == 0 or (self.n_one[l] != self.n_one[l - 1]) or (
                         self.n_two[l] != self.n_two[l - 1]):
-                    one_electron_tmp[:, i, :] = torch.tanh(self.v[l](f.to(
-                        torch.float32)))
+                    one_electron_tmp[:, i, :] = torch.tanh(self.v[l](f))
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:5014]: Can you explain the changes in this section?

File: deepchem/models/torch_models/layers.py [hunk 4]
```diff
@@ -5156,7 +5159,11 @@ def forward(self, one_electron: torch.Tensor,
                                    dim=2))) * self.pi[one_d_index].T,
                                   dim=1)
 
-        return psi_up, psi_down
+            d_down = torch.det(psi_down[:, k, :, :].clone())
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:5162]: What is the purpose of these changes? Can you add a comment?

**Review by rbharath — COMMENTED** [review]
Have several requests for comments below

File: deepchem/models/tests/test_layers.py (hunk 1 above)
Comment (shaipranesh2) [reply deepchem/models/tests/test_layers.py]: Removed double datatypes as per our previous conversation. (Thought double datatypes will improve precision, but slowed down the model)

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (shaipranesh2) [reply deepchem/models/torch_models/layers.py:4945]: Removed double datatypes as per our previous conversation. (Thought double datatypes will improve precision, but slowed down the model)

File: deepchem/models/torch_models/layers.py (hunk 3 above)
Comment (shaipranesh2) [reply deepchem/models/torch_models/layers.py:5014]: Here the tensor was already in float32, so ".to(torch.float32)" adds redundancy

File: deepchem/models/torch_models/layers.py (hunk 4 above)
Comment (shaipranesh2) [reply deepchem/models/torch_models/layers.py:5161]: I changed the layer to return the determinant value of orbital matrix and the wavefunction's value. (the determinant calculation when clubbed with this layer, avoids redundant loops in the code)

**Review by rbharath — APPROVED** [review]
LGTM
